Source/WebCore/loader/FetchPriority.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2023 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "FetchPriority.h"
+
+namespace WebCore {
+
+FetchPriority parseFetchPriority(StringView fetchPriorityString)
+{
+    if (equalLettersIgnoringASCIICase(fetchPriorityString, "high"_s))
+        return FetchPriority::High;
+    if (equalLettersIgnoringASCIICase(fetchPriorityString, "low"_s))
+        return FetchPriority::Low;
+    return FetchPriority::Auto;
+}
+
+String fetchPriorityToString(const FetchPriority& fetchPriority)
+{
+    switch (fetchPriority) {
+    case FetchPriority::High:
+        return "high"_s;
+    case FetchPriority::Low:
+        return "low"_s;
+    case FetchPriority::Auto:
+        return "auto"_s;
+    }
+    ASSERT_NOT_REACHED();
+    return { };
+}
```

Comment:
This code is probably generated by the binding generator.
We could either use RequestPriority everywhere or include JSRequestPriority.h and add the necessary WebIDL key words to generate parsing/serialization of the enum.

Reply:
Done. I could see only one fetch data structure where we did this before, looks like there is room to do more (for ReferrerPolicy the parsing is likely duplicated.